build/root/usr/local/bin/unentitle.sh

```diff
@@ -0,0 +1,8 @@
+#! /bin/bash -e
+
+set -o pipefail
+set -o errexit
+set -o nounset
+set -x
+
+toolbox/entitlement/undeploy.sh 
```

Comment:
why not calling ` toolbox/entitlement/undeploy.sh ` directly?

and I think you'll need to wait for the cluster to be unentitled, one way or another
one way could be to scale down to `0` the # of GPU nodes, unentitle, then scale up to `1`,
this way you know that the GPU node will come up without the entitlement files


another way is to pin the testing pod to nodes with `nvidia.com/gpu.present=true`, and wait until the entitlement test fails

Reply:
(this file is never called BTW)

Reply:
> why not calling toolbox/entitlement/undeploy.sh directly?

Wasn't sure where the toolbox is mapped exactly relative to the CI working directory

>one way could be to scale down to 0 the # of GPU nodes, unentitle, then scale up to 1,
this way you know that the GPU node will come up without the entitlement files
Yeah I guess that if I do that I don't have to use undeploy at all

Reply:
> Wasn't sure where the toolbox is mapped exactly relative to the CI working directory

yes, there are things unclear in this regard, but the implicit assumption is that `run` and `ci-entrypoint_*` all run with `CWD=$HOME`, never change directory, and `toolbox/...` is available from there

Reply:
> and I think you'll need to wait for the cluster to be unentitled, one way or another
Maybe `toolbox/entitlement/undeploy.sh` playbook should do this waiting. What do you think?

Reply:
the thing is that it's hard to know how long to wait, eg,
a) do you wait for all the nodes to be rebooted? 
b) do you wait for one random node to be rebooted (that what we do for `toolbox/entitlement/wait.sh` as my goal is to validate that the entitlement is working, not to wait for all the nodes to be entitled)

maybe  `toolbox/entitlement/wait.sh` could be easily updated to wait for a pod failing the entitlement test (case `b)` above,
but that's not a full solution
or maybe it can also receive a node name, to specify which node we want to test
```
toolbox/entitlement/wait.sh --not-entitled --node=my_node
```
(mind that node "name != node hostname" for the node selector)

Reply:
You can wait for machineconfigpool status:

After removing machineconfig:
![image](https://user-images.githubusercontent.com/10882062/122100513-90902b80-ce13-11eb-80d0-e3ef4b0923af.png)

After the removal is finally complete:
![image](https://user-images.githubusercontent.com/10882062/122101515-af42f200-ce14-11eb-975b-7ec0be1c4c6f.png)

Reply:
@omertuc yes indeed that can be an easy way to wait for a full removal of the the MachineConfigs,
then it can become a flag of `undeploy.sh`:
```
toolbox/entitlement/undeploy.sh --wait
```